src/main/java/io/debezium/connector/jdbc/transform/ConvertCloudEventToSaveableForm.java

```diff
@@ -0,0 +1,155 @@
+ * CloudEvent's `data` field is flattened if needed
+ *
+ * @author Roman Kudryashov
+ */
+public class ConvertCloudEventToSaveableForm implements Transformation<SinkRecord> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConvertCloudEventToSaveableForm.class);
+
+    private static final String FIELD_NAME_SEPARATOR = ":";
+
+    private static final String CLOUD_EVENTS_SCHEMA_NAME_SUFFIX = ".CloudEvents.Envelope";
+
+    private static final Field FIELDS_MAPPING = Field.create("fields.mapping")
+            .withDisplayName("Specifies a list of pairs with mappings between a CloudEvent's fields and names of database columns")
+            .withType(ConfigDef.Type.LIST)
+            .withValidation(AdditionalFieldsValidator::isListOfStringPairs)
+            .withWidth(ConfigDef.Width.MEDIUM)
+            .withImportance(ConfigDef.Importance.HIGH)
+            .withDescription("Specifies a list of pairs with mappings between a CloudEvent's fields and names of database columns");
+
+    private Map<String, String> fieldsMapping;
+
+    private final JsonConverter jsonDataConverter = new JsonConverter();
+
+    @Override
+    public ConfigDef config() {
+        final ConfigDef config = new ConfigDef();
+        Field.group(config, null, FIELDS_MAPPING);
+        return config;
+    }
+
+    @Override
+    public void configure(final Map<String, ?> configs) {
+        final Configuration config = Configuration.from(configs);
+
+        final List<String> rawFieldsMapping = config.getList(FIELDS_MAPPING);
+        fieldsMapping = Collections.unmodifiableMap(parseFieldsMapping(rawFieldsMapping));
+
+        Map<String, Object> jsonDataConverterConfig = new HashMap<>();
+        jsonDataConverterConfig.put(JsonConverterConfig.SCHEMAS_ENABLE_CONFIG, false);
+        jsonDataConverterConfig.put(JsonConverterConfig.TYPE_CONFIG, "value");
+        jsonDataConverter.configure(jsonDataConverterConfig);
+    }
+
+    private Map<String, String> parseFieldsMapping(List<String> rawFieldsMapping) {
+        final Map<String, String> parsedFieldsMapping = new HashMap<>();
+        for (String rawFieldMapping : rawFieldsMapping) {
+            final String[] parts = rawFieldMapping.split(FIELD_NAME_SEPARATOR);
+            final String cloudEventFieldName = parts[0];
+            final String databaseColumnName;
+            if (rawFieldMapping.contains(FIELD_NAME_SEPARATOR)) {
+                databaseColumnName = parts[1];
+            }
+            else {
+                databaseColumnName = cloudEventFieldName;
+            }
+            parsedFieldsMapping.put(cloudEventFieldName, databaseColumnName);
+        }
+        return parsedFieldsMapping;
+    }
+
+    @Override
+    public SinkRecord apply(final SinkRecord record) {
+        if (!record.valueSchema().name().endsWith(CLOUD_EVENTS_SCHEMA_NAME_SUFFIX) || fieldsMapping.isEmpty()) {
```

Comment:
```suggestion
        if (record == null || !record.valueSchema().name().endsWith(CLOUD_EVENTS_SCHEMA_NAME_SUFFIX) || fieldsMapping.isEmpty()) {
```
Guard against tombstone messages